ardrive_network/lib/src/responses.dart

```diff
@@ -0,0 +1,35 @@
+import 'package:equatable/equatable.dart';
+
+class ArDriveNetworkResponse {
+  ArDriveNetworkResponse({
+    required this.data,
+    this.statusCode,
+    this.statusMessage,
+    required this.retryAttempts,
+  }) {
+    this.data = data;
+    this.statusCode = statusCode;
+    this.statusMessage = statusMessage;
+    this.retryAttempts = retryAttempts;
+  }
+
+  dynamic data;
+  int? statusCode;
+  String? statusMessage;
+  int retryAttempts;
+}
+
+class ArDriveNetworkException extends Equatable implements Exception {
+  final int retryAttempts;
+  final Object dioException;
+
+  ArDriveNetworkException(
+      {required this.retryAttempts, required this.dioException});
+  @override
+  List<Object?> get props => [this.retryAttempts];
```

Comment:
Should it include dioException? If not then two ArDriveNetworkExceptions with the same retryAttempts value would be equal
```suggestion
  List<Object?> get props => [retryAttempts, dioException];
```

Reply:
It will throw dioException, but we will only compare (Equatable) the retryAttempts for test purposes.

Reply:
For testing purposes only. Got ya!